MapboxVision/Services/VideoSource/ObservableVideoSource.swift

```diff
@@ -4,38 +4,67 @@ import Foundation
  Helper class handling observers: storing, releasing, notifying.
  Observers are held weakly by the instance of the class.
  You may inherit your video source from this class to avoid handling observers yourself.
+
+ - Important
+ A deadlock can occur when add/remove methods are called inside the notify's closure.
```

Comment:
`os_unfair_lock` crashes on recursive execution so it won't let a deadlock to occur, as I understand.
Let's just mention that we're using such type of lock and warn from including `add` and `remove` in `notify`
I'd add this note to `notify` doc too.

Reply:
You're right. According to https://opensource.apple.com/source/libplatform/libplatform-177.250.1/include/os/lock.h.auto.html

```
 * If the lock is unlocked or owned by a different thread, this function
 * asserts and terminates the process
```

Updated the doc comments for class/methods.